src/app/generic-listing-page/generic-listing-page.component.ts

```diff
@@ -76,4 +96,11 @@ export class GenericListingPageComponent  implements OnInit {
     this.count = 0
   }
 
+  handleInput(event: any) {
+    this.searchTerm = event.target.value;
+    this.page = 1;
```

Comment:
the the below 3 line u can call reset() function

Reply:
done